craft_parts/infos.py

```diff
@@ -0,0 +1,209 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Project, part and step information classes."""
+
+import logging
+import platform
+from typing import Any, Dict, List, Optional
+
+from craft_parts import errors, utils
+from craft_parts.dirs import ProjectDirs
+
+logger = logging.getLogger(__name__)
+
+
+class ProjectInfo:
+    """Project-level information containing project-specific fields.
+
+    :param application_name: A unique identifier for the application using
+        Craft Parts.
+    :param arch: The architecture to build for. Defaults to the host system
+        architecture.
+    :param parallel_build_count: The maximum number of concurrent jobs to be
+        used to build each part of this project.
+    :param plugin_version: The plugin API version. Currently only ``v2`` is
```

Comment:
Consider dropping the notion of plugin version until we have a need for it.  When we do need to start versioning things, perhaps we can/would do it differently (e.g. not at a project level).

Reply:
I'll remove this argument from the user-exposed API and keep it internal until we have a new plugin version.